elementary/messages/messaging_integrations/slack_web.py

```diff
@@ -0,0 +1,160 @@
+from elementary.messages.messaging_integrations.base_messaging_integration import (
+    BaseMessagingIntegration,
+    MessageSendResult,
+)
+from elementary.messages.messaging_integrations.exceptions import (
+    MessagingIntegrationError,
+)
+from elementary.tracking.tracking_interface import Tracking
+from elementary.utils.log import get_logger
+
+logger = get_logger(__name__)
+
+ONE_MINUTE = 60
+ONE_SECOND = 1
+
+
+Channel = NewType("Channel", str)
+
+
+class SlackWebMessageContext(BaseModel):
+    id: str
+    channel: Channel
+
+
+class SlackWebMessagingIntegration(
+    BaseMessagingIntegration[Channel, SlackWebMessageContext]
+):
+    def __init__(self, client: WebClient, tracking: Optional[Tracking] = None) -> None:
+        self.client = client
+        self.tracking = tracking
+        self._email_to_user_id_cache: Dict[str, str] = {}
+
+    @classmethod
+    def from_token(
+        cls, token: str, tracking: Optional[Tracking] = None
+    ) -> "SlackWebMessagingIntegration":
+        client = WebClient(token=token)
+        client.retry_handlers.append(RateLimitErrorRetryHandler(max_retry_count=5))
+        return cls(client, tracking)
+
+    def supports_reply(self) -> bool:
+        return True
+
+    def send_message(
+        self, destination: Channel, body: MessageBody
+    ) -> MessageSendResult[SlackWebMessageContext]:
+        formatted_message = format_block_kit(body, self.get_user_id_from_email)
+        return self._send_message(destination, formatted_message)
+
+    def reply_to_message(
+        self,
+        destination: Channel,
+        message_context: SlackWebMessageContext,
+        body: MessageBody,
+    ) -> MessageSendResult[SlackWebMessageContext]:
+        formatted_message = format_block_kit(body, self.get_user_id_from_email)
+        return self._send_message(
+            destination, formatted_message, thread_ts=message_context.id
+        )
+
+    @sleep_and_retry
+    @limits(calls=1, period=ONE_SECOND)
+    def _send_message(
+        self,
+        destination: Channel,
+        formatted_message: FormattedBlockKitMessage,
+        thread_ts: Optional[str] = None,
+    ) -> MessageSendResult[SlackWebMessageContext]:
+        try:
+            response = self.client.chat_postMessage(
+                channel=destination,
+                blocks=json.dumps(formatted_message["blocks"]),
+                attachments=json.dumps(formatted_message["attachments"]),
+                thread_ts=thread_ts,
+            )
+        except SlackApiError as e:
+            self._handle_send_err(e, destination)
+            return self._send_message(destination, body, thread_ts)
+
+        return MessageSendResult(
+            message_context=SlackWebMessageContext(
+                id=response["ts"], channel=response["channel"]
+            ),
+            timestamp=response["ts"],
+        )
+
+    def _handle_send_err(self, err: SlackApiError, channel_name: str) -> bool:
```

Comment:
from this point down, mainly adapted from the existing clients, with some tweaks